spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1487,11 +1485,8 @@ def create_new_client(self, give_focus=True, filename='', is_cython=False,
                               ask_before_closing=ask_before_closing,
                               css_path=self.css_path,
                               configuration=self.CONFIGURATION,
-                              handlers=self.registered_spyder_kernel_handlers)
-
-        # Change stderr_dir if requested
-        if self._test_dir:
-            client.stderr_dir = self._test_dir
+                              handlers=self.registered_spyder_kernel_handlers,
+                              std_dir=self._test_dir)
```

Comment:
```suggestion
                              handlers=self.registered_spyder_kernel_handlers)
```

I think this is not necessary here.